### PR TITLE
feat: Initial Welcome page

### PR DESCRIPTION
**Pull request by deboer-tim**
### What does this PR do?

Creates a new Welcome page that will be shown once for every user. As per the feature this is just a simple page for now, but could be expanded in future sprints to help the user install podman or do other first-time setup.

Implementation notes:
- A welcome.version configuration property (not shown in settings) is used to store the fact that the welcome page has been shown. The value is set to 'initial' for now. My intention was that this could be set to the version of podman desktop or welcome page shown so in a future sprint we can decide when people who have seen the welcome should now see a newer/better version.
- The new page is an overlay that appears only once, based on the configuration.
- I've added the podman desktop logo as images/DesktopIcon because I don't believe we've used it elsewhere, but will likely need it on other pages in the future.
- The welcome page has a blurred background image and a static list of supported container engines. I've added a link to Settings and made a few other minor changes based on feedback from previous draft PR #1566.

### Screenshot/screencast of this PR

Screenshot 2023-03-16 at 12 10 23 PM

### What issues does this PR fix or reference?

Issue #1334.

### How to test this PR?

The welcome should appear the first time you start and then never again. To see the welcome a second time, remove the welcome.version property from the configuration (e.g. .local/share/containers/podman-desktop/configuration/settings.json on Mac) or delete the config entirely.

### CODE REVIEW

File: packages/renderer/src/lib/welcome/WelcomePage.svelte
Comment (benoitf) [inline packages/renderer/src/lib/welcome/WelcomePage.svelte]:
```suggestion
  if (!ver) {
```

File: packages/renderer/src/lib/welcome/WelcomePage.svelte
Comment (benoitf) [inline packages/renderer/src/lib/welcome/WelcomePage.svelte:16]: From what I understand from your description of the PR
I guess we should store the version of Podman Desktop so it'll easy to know later what to while if we store 'initial' we may never know what to do later ?

File: packages/renderer/src/lib/welcome/WelcomePage.svelte
Comment (benoitf) [inline packages/renderer/src/lib/welcome/WelcomePage.svelte]:
```suggestion
      <div class="flex justify-center text-lg font-bold p-4"><span class="mr-2">🎉</span>Welcome to Podman Desktop!</div>
```
just a suggestion, not mandatory maybe want to use tailwind there ?

File: packages/renderer/src/lib/welcome/WelcomePage.svelte
Comment (benoitf) [inline packages/renderer/src/lib/welcome/WelcomePage.svelte:61]: for on:click we need to use `<button>` instead of `<div>` to avoid any a11y issues

also it's easier to test the component with a component test

File: packages/renderer/src/lib/welcome/WelcomePage.svelte
Comment (benoitf) [inline packages/renderer/src/lib/welcome/WelcomePage.svelte]: we probably need a component test of WelcomePage

we have tests for WelcomeUtils but would be nice to test that if we have buttons that can be clicked and that they're doing something (redirecting people)

**Review by benoitf — COMMENTED** [review]
looks like I got a scrollbar on the right by default

https://user-images.githubusercontent.com/436777/225687418-6fc58133-c68b-473c-9e2e-b86d830ce05d.mp4

if I increase width/height it's gone but I have the right scrollbar on every launch

it's visible with height = 600px, gone after height > 602px

File: packages/renderer/src/lib/welcome/WelcomePage.svelte
Comment (deboer-tim) [reply packages/renderer/src/lib/welcome/WelcomePage.svelte:16]: I thought of using the Podman Desktop version too, but if we didn't change this for a sprint or two I don't think we want to show it again every time users upgrade. I could use a different string like "v1" if you prefer.

File: packages/renderer/src/lib/welcome/WelcomePage.svelte
Comment (benoitf) [inline packages/renderer/src/lib/welcome/WelcomePage.svelte]: There is no tailwind properties for that ?

Comment (benoitf) [reply packages/renderer/src/lib/welcome/WelcomePage.svelte]: class="bg-[url('/img/hero-pattern.svg')]"

Comment (benoitf) [reply packages/renderer/src/lib/welcome/WelcomePage.svelte]: https://tailwindcss.com/docs/background-repeat

bg-no-repeat

File: packages/renderer/src/lib/welcome/WelcomePage.svelte
Comment (benoitf) [reply packages/renderer/src/lib/welcome/WelcomePage.svelte:16]: AFAIK if user upgrade the property is still stored so we won't show again the welcome page.
But then we could say that we need to display something on an upgrade while if we always store `initial` we don't really know 

It's a suggestion, not blocking the PR

**Comment by mairin** [comment]
@deboer-tim This looks amazing and very accurate / pixel-perfect to the mockups from the screenshot. I grabbed a checkout of the PR to try it out "in person" but I'm not sure how to trigger it to popup? I don't know how to interact with the configuration registry.

**Comment by mairin** [comment]
@benoitf got me past my issues getting it launched (thank you :) )

So on Linux by default it is launching a bit small / cut-off on the bottom (fairly, in debug mode) here's how it looks on a fresh checkout + yarn + yarn watch with no config file on disk:

Screenshot from 2023-03-16 18-46-30

This is an issue because the primary button to dismiss the window / move forward is hidden by default. I think maybe what would address that is setting useContentSize==true when you create the window (see https://www.electronjs.org/docs/latest/api/browser-window) but I am not sure. 

The vertical scroll here too I think should just never appear - well at least from a Linux desktop user POV - something like this is really a splash window and you can resize the window using the window manager's drag handles so you dont need the scroll. In the browser developer tool i think this is the right way to do it, ymmv

Screenshot from 2023-03-16 18-59-12


Screenshot from 2023-03-16 18-47-35

Hope this is helpful, I am really excited about how it's turning out :)

File: packages/renderer/src/lib/images/DesktopIcon.svelte
Comment (jeffmaury) [inline packages/renderer/src/lib/images/DesktopIcon.svelte]: Are we sure this URL is the good one ?

File: packages/renderer/src/lib/welcome/WelcomePage.svelte
Comment (jeffmaury) [reply packages/renderer/src/lib/welcome/WelcomePage.svelte:16]: I would suggest the value stored should be the signature or last commit id of the welcome page

File: packages/renderer/src/lib/welcome/WelcomePage.svelte
Comment (deboer-tim) [reply packages/renderer/src/lib/welcome/WelcomePage.svelte]: > class="bg-[url('/img/hero-pattern.svg')]"

This works for static values, but not for svelte imported/dynamic values - see https://v2.tailwindcss.com/docs/just-in-time-mode#arbitrary-value-support. Leave as-is, or use an external url for the image?

I used the style for no repeat to keep the background 'in one place', I will switch to Tailwind.

**Comment by deboer-tim** [comment]
> ...
> This is an issue because the primary button to dismiss the window / move forward is hidden by default. I think maybe what would address that is setting useContentSize==true when you create the window (see https://www.electronjs.org/docs/latest/api/browser-window) but I am not sure.
> ... 
> The vertical scroll here too I think should just never appear - well at least from a Linux desktop user POV - something like this is really a splash window and you can resize the window using the window manager's drag handles so you dont need the scroll. In the browser developer tool i think this is the right way to do it, ymmv

100% agreed that by default the bottom bar/close button should be visible and no scroll bar.

Currently the window opens at 1050x600 and minimum size is 640x600. It looks ok on my Mac, but presumably the rendering differences mean it is a little bigger than 600px vertical for you. 

useContentSize isn't making any difference for me (presumably b/c there's a loading screen first?), so the other options are either trying to reduce the vertical size of the welcome screen (I've already tried to squeeze padding as much as possible) or increase the default window height. I don't love increasing our default screen size, but I think the size is ok and will work better for future onboarding screens so I'll change the height to 700px here: https://github.com/containers/podman-desktop/blob/4d890cd9bac70d666dd32b66662bfdfb6ffa5a3b/packages/main/src/mainWindow.ts#L32

I don't want to increase the absolute minimum size, since I frequently resize Podman Desktop to 900x600px-ish. This means that users could still decrease the size of the onboarding window and hide content, and without a scroll bar there would be no way to get to it or even know that you can't see something. The current overflow-y-auto should only show the scrollbar when needed, and I can move it to the centre content area so that the titlebar and footer bar are always visible.

File: packages/renderer/src/lib/images/DesktopIcon.svelte
Comment (cdrage) [reply packages/renderer/src/lib/images/DesktopIcon.svelte]: save locally svg perhaps?

File: packages/renderer/src/lib/welcome/WelcomePage.svelte
Comment (deboer-tim) [reply packages/renderer/src/lib/welcome/WelcomePage.svelte]: Fixed in second commit.

File: packages/renderer/src/lib/welcome/WelcomePage.svelte
Comment (deboer-tim) [reply packages/renderer/src/lib/welcome/WelcomePage.svelte]: Fixed in second commit.

File: packages/renderer/src/lib/welcome/WelcomePage.svelte
Comment (deboer-tim) [reply packages/renderer/src/lib/welcome/WelcomePage.svelte:61]: Fixed in second commit.

File: packages/renderer/src/lib/welcome/WelcomePage.svelte
Comment (deboer-tim) [reply packages/renderer/src/lib/welcome/WelcomePage.svelte]: Added component test in second commit.

File: packages/renderer/src/lib/welcome/WelcomePage.svelte
Comment (deboer-tim) [reply packages/renderer/src/lib/welcome/WelcomePage.svelte]: See second commit. I moved to bg-no-repeat but left the dynamic url for now because using Svelte {bgImage} felt cleaner than Tailwind url, but this is a judgement call so let me know if you disagree.

File: packages/renderer/src/lib/images/DesktopIcon.svelte
Comment (deboer-tim) [reply packages/renderer/src/lib/images/DesktopIcon.svelte]: Doh, good catch. See second commit, I included the png and fixed the reference.

**Comment by deboer-tim** [comment]
Second commit fixes the majority of review comments.

> I would suggest the value stored should be the signature or last commit id of the welcome page

I cannot find any clean way to do this (e.g. see https://stackoverflow.com/questions/3442874/in-git-how-can-i-write-the-current-commit-hash-to-a-file-in-the-same-commit) that wouldn't be uglier than using a constant, but let me know if you have any other ideas.

**Review by benoitf — APPROVED** [review]
unit tests and component tests 🎉 

it works on my side
image

probably before merging we should see if it's ok on Windows and Linux

**Review by jeffmaury — APPROVED** [review]
Looks ok on Windows

**Review by lstocchi — APPROVED** [review]
Ok on Fedora 37 👍

**Pull request merged**
